src/models/scatter.js

```diff
@@ -455,7 +455,7 @@ nv.models.scatter = function() {
                     return 'translate(' + nv.utils.NaNtoZero(x(getX(d[0],d[1]))) + ',' + nv.utils.NaNtoZero(y(getY(d[0],d[1]))) + ')'
                 })
                 .remove();
-            points.filter(function (d) { return scaleDiff || sizeDiff || getDiffs(d, 'x', 'y'); })
+            points.filter(function (d) { return scaleDiff || sizeDiff || getDiffs(d, 'x', 'y') || getDiffs(d, 0, 1); })
```

Comment:
Ah yes, good shout. I think I was trying to prevent layering up unnecessary function calls when I first added this (after all it's supposed to be an optimisation). But I did not notice the scoped getX/Y has a public setter on 607/8... besides I've now added another function call anyway.